### PR TITLE
refactor: add Avro Schema Parser

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR does the following:

- Ports https://github.com/asyncapi/avro-schema-parser to TypeScript.
- Makes it compliant with the [new interface](https://github.com/asyncapi/parser-js/issues/480) for Schema Parsers (new `validate()` method).

> **Note**
> There are some TODOs I left in purpose to have a discussion.

**Related issue(s)**
https://github.com/asyncapi/parser-js/issues/480

### CODE REVIEW

File: src/schema-parser/avro-schema-parser.ts
Comment (smoya) [inline src/schema-parser/avro-schema-parser.ts:36]: The big change VS what we had previously is that the new interface passes directly the Schema in `input.data` rather than the message object. 
I decided to add the message object in the `input.meta` and only in case it is there, do the modifications (if they are being done here finally).

**Comment by derberg** [comment]
I'm not involved in rewriting to TypeScript, so this might be silly question, but why are you porting avro schema parser here inside the parser?

**Comment by smoya** [comment]
> I'm not involved in rewriting to TypeScript, so this might be silly question, but why are you porting avro schema parser here inside the parser?

Even though there are unit tests, the idea is to have everything here, to do proper testing of the parser with those schema parsers, and once we confirm everything works as expected, to move the code to each schema parser repo.

**Review by fmvilas — APPROVED** [review]
LGTM 👍 Pinging @dalelane and @M3lkior as they're both code owners of the Avro schema parser repo.

File: src/schema-parser/avro-schema-parser.ts
Comment (magicmatatjahu) [reply src/schema-parser/avro-schema-parser.ts:52]: Now you can leave that logic, I will remember about it.

File: src/schema-parser/avro-schema-parser.ts
Comment (magicmatatjahu) [inline src/schema-parser/avro-schema-parser.ts:345]: `parse` function will only run when given schema is valid, so it's not needed :)

File: src/schema-parser/avro-schema-parser.ts
Comment (magicmatatjahu) [inline src/schema-parser/avro-schema-parser.ts:146]: What do you mean by `we lose boolean type?`? Do you mean that JSON Schema can be true or false? I don't think so that it's a problem here.

**Review by magicmatatjahu — COMMENTED** [review]
Great! I left some comments :) Please refer to them and we can merge it :)

File: src/schema-parser/avro-schema-parser.ts
Comment (magicmatatjahu) [reply src/schema-parser/avro-schema-parser.ts:36]: Ok, I will remember about that :) Avro parser there is a bit problematic 😅

File: src/schema-parser/avro-schema-parser.ts
Comment (smoya) [reply src/schema-parser/avro-schema-parser.ts]: It's done now!

**Review by magicmatatjahu — APPROVED** [review]
LGTM! 🚀 Should we wait for review of parser's maintainers?

**Review by dalelane — APPROVED** [review]

**Comment by smoya** [comment]
We can merge now as we have two approvals from codes owner.

**Pull request merged**
